uxarray/grid/grid.py

```diff
@@ -2485,3 +2498,89 @@ def get_faces_between_latitudes(self, lats: Tuple[float, float]):
 
         """
         return faces_within_lat_bounds(lats, self.face_bounds_lat.values)
+
+    def get_faces_containing_point(self, point, tolerance=ERROR_TOLERANCE):
```

Comment:
Can we slightly modify the structure here to support the following parameters
- point_xyz (Default None)
- point_lonlat (Default None)

The user **_needs_** to provide at least one of these, which would mean the logic below where we convert one of them to xyz or lonlat respectively can remain, however if they provide both we wouldn't need to do any extra conversion.

For many of our grids, we already have both the Spherical and Cartesian coordinate loaded, so we can leverage this. 

This reiterates @hongyuchen1030 's points about trying to avoid as many of these conversions, since they can lead to precision loss and trig functions are not the cheapest to run.